src/devices/controller/controller.cpp

```diff
@@ -109,7 +146,196 @@ void loop() {
 
   // Update "sending" status LED.
   leds[kStatusRight] = CRGB(0, 0, broadcast_led_timer.Active() ? 255 : 0);
-  FastLED.show();
+}
+
+void RunColorMode() {
+  for (uint8_t i = 0; i < 36; i++) {
+    if ((i % 12) > 4 && (i % 12) <= 6) {
+      SetMainLed(i, CRGB(0, 0, 0));
+    } else {
+      SetMainLed(i, colors[(i / 6)]);
+    }
+  }
+
+  bool pressed = false;
+  uint8_t color_index = 0;
+  if (left_buttons.Button1Pressed()) {
+    pressed = true;
+    color_index = 0;
+  } else if (left_buttons.Button2Pressed()) {
+    pressed = true;
+    color_index = 2;
+  } else if (left_buttons.Button3Pressed()) {
+    pressed = true;
+    color_index = 4;
+  } else if (right_buttons.Button1Pressed()) {
+    pressed = true;
+    color_index = 1;
+  } else if (right_buttons.Button2Pressed()) {
+    pressed = true;
+    color_index = 3;
+  } else if (right_buttons.Button3Pressed()) {
+    pressed = true;
+    color_index = 5;
+  }
+
+  if (left_buttons.Button1Pressed()) {
+    SetLeftButtonLeds(kButtonPressedBrightness, kButtonActiveBrightness,
+                      kButtonActiveBrightness);
+  } else if (left_buttons.Button2Pressed()) {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonPressedBrightness,
+                      kButtonActiveBrightness);
+  } else if (left_buttons.Button3Pressed()) {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                      kButtonPressedBrightness);
+  } else {
+    SetLeftButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                      kButtonActiveBrightness);
+  }
+
+  if (right_buttons.Button1Pressed()) {
+    SetRightButtonLeds(kButtonPressedBrightness, kButtonActiveBrightness,
+                       kButtonActiveBrightness);
+  } else if (left_buttons.Button2Pressed()) {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonPressedBrightness,
+                       kButtonActiveBrightness);
+  } else if (left_buttons.Button3Pressed()) {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonPressedBrightness);
+  } else {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonActiveBrightness);
+  }
+
+  SetBottomButtonLeds(0, 0, 0);
 
+  if (pressed) {
```

Comment:
Is this debounced? Does the call below block?

Reply:
Yes, `AnalogButton` is debounced (although v1.1 will eliminate it). `state_machine.SetEffect(&control_packet)` does block briefly (about 1ms).